taxonomy/definitions.md

```diff
@@ -0,0 +1,21 @@
+# Glossary of Terms: Understanding Lifecycle Phases in Technology
+
+This document aims to clearly define key terms related to the lifecycle of software and hardware products, providing a reference for understanding various phases from sales or creation to support termination. 
+
+**Note**: With open source software or hardware, there is typically no "end-of-sales".
+
+## 1. Software End-of-Life (SEoL) or Software End-of-Sales (SEoS)
+
+**Definition:** Software End-of-Life refers to the stage in a software product's lifecycle where it is no longer sold. 
+
+## 2. Hardware End-of-Life (HEoL) or Hardware End-of-Sales (HEoS)
+
+**Definition:** Hardware End-of-Life signifies the phase where a piece of hardware (like a server, router, or computer component) is no longer sold by its manufacturer.
+
+## 3. End-of-Software Maintenance or Security Vulnerability Support
+
+**Definition:** This term indicates the final phase of a software product’s lifecycle during which the developer ceases to provide maintenance services, including security patches, bug fixes, or any other form of software updates. Beyond this point, the software may become vulnerable to security risks as new threats emerge. However, a vendor may still support technical questions and may also investigate vulnerabilities in their products without providing a fix.
+
+## 4. Last Date of Support (End-of-Support)
+
+**Definition:** The Last Date of Support, or End-of-Support, refers to the final date when all types of support, including technical, maintenance, and security updates, will be provided for a software or hardware product. After this date, the product is considered unsupported and may pose risks or operational inefficiencies for users.
```

Comment:
A new item/definition that we should add is "Extended Support" as suggested in https://github.com/oasis-tcs/openeox/issues/15

Reply:
This defination mostly seen on Operating System, taking a try .. please review/update.
Term : End of Extended Support
Definition : Extended Support period is when  only critical security are provided under special agreements and after End of Extended Support date  there is no support is provided.

Reply:
Similar to Extended Support, different segment may have a specialized term/definition. Can we allow Additional definition where a company/vendor/partner can put a custom information as text, URL ?

Reply:
> Similar to Extended Support, different segment may have a specialized term/definition. Can we allow Additional definition where a company/vendor/partner can put a custom information as text, URL ?

To be frank: That is a bad idea. A standard should cover most (if not all) use cases. Introducing a way to use own definitions will actually prevent people from leaving their old system behind and following the standard. As user, I want to have **one** dataset that is comparable, not hundreds and being forced to figure out the mapping between the companies definition and OpenEoX...

Reply:
> This defination mostly seen on Operating System, taking a try .. please review/update. Term : End of Extended Support Definition : Extended Support period is when only critical security are provided under special agreements and after End of Extended Support date there is no support is provided.

I had the same idea. Maybe, we need to differentiate between **standard** and **extended** support? Or paid/unpaid?

Reply:
I do agree with @tschmidtb51 that arbitrary definitions can lead to problems.

But there are products out there with at least three different "levels" of support and I suspect others might have more.
So, I do believe that the standard should allow for "a list of support definitions" the interpretation of which can be up to the user.

e.g. RedHat has "Full Support", "Maintenance Support", "Extended Update Support"
Ubuntu has something similar.

We need to be able to model those kinds of things and I don't think the standard should mandate how many levels are allowed.
"paid/unpaid" -> this seems more like an attribute on a support level.

Reply:
I agree. Unfortunately, sometimes the tools (i.e., CMDB, vuln mgmt, scanners) will be the ones making the "interpretation" of these definitions, so we may need to find a balance.